### PR TITLE
Static Hash Finetuning

### PR DESCRIPTION
**Pull request by l1npengtul**

### CODE REVIEW

**Comment by schungx** [comment]
@l1npengtul  this fails on CI.  I think a few syntax errors.

**Comment by l1npengtul** [comment]
im done for today... other things to work on...

ill fix it tomorrow :p

**Comment by schungx** [comment]
One more nit-pick:  Function `get_or_init` always returns `Some`.  Therefore, there is really no reason for it to return `Option<T>`.  It should simply return `T` or `&T`.

File: src/config/hashing.rs [hunk 1]
```diff
@@ -136,9 +136,9 @@ where
     }
 
     #[must_use]
-    pub fn get_or_init(&self, f: impl FnOnce() -> T) -> Option<T> {
-        let value = f();
+    pub fn get_or_init(&self, f: impl FnOnce() -> T) -> Option<&'static T> {
```
Comment (schungx) [inline src/config/hashing.rs:139]: `get_or_init` alreays return `Some` so it should be simply `&'static T`.

Comment (l1npengtul) [reply src/config/hashing.rs:139]: This can still be null since the AHASH_SEED envvar isnt always set

**Comment by l1npengtul** [comment]
Ok IDK what happened with the diff...

File: src/config/hashing.rs (hunk 1 above)
Comment (schungx) [reply src/config/hashing.rs:139]: Well, `T` itself is `Option`, so if `AHASH_SEED` isn't set, it'll return `&'static None`.

Comment (schungx) [reply src/config/hashing.rs:139]: No worries, I'll just merge it and fiddle with it later.

**Comment by l1npengtul** [comment]
yeah

**Pull request merged**